### DISCUSSION TITLE
Laravel Swoole

### DISCUSSION THREAD

**Discussion by Spartaques** [laravel/framework#42558] [Ideas] [answered]
Hello guys. For now php is blocking and laravel is blocking and all this shit is blocking. It's very comfortable to work with Laravel and all ecosystem, but when the question is about performance, high throughtput, etc... it becomes not comfortable).
I like laravel and coding using it more then 6 years. I'm an software architect of start up company.
We pick laravel when we start to do our project, but for now we have 4 services, 1 of them is on node.js and 2 in php swoole fully compatible framework called hyperf https://hyperf.wiki/2.2/. So, i can tell you that it just not work. All IO operations in laravel is blocking, so performance is very bad and cannot be comparad to exact task performance in nodejs or hyperf swoole.
So, do you guys have any plan to make laravel fully compatible with swoole cooroutine, or you decide to not do this? I know some company that fork symfony or laravel and doing custom framework that compatible with swoole, and it works in some cases x1000 faster.

For me, IO blocking becomes a big problem and big companies just not use solutions that don't support async io operatios. So, i want to know if i must implement all this stuff myself or you guys have a plan for this in near future?

**Comment by Spartaques** [comment]
All other languages top frameworks, such as django, spring, except ruby on rails is fully or partly compatible with async io

**Comment by mahdiaslami** [comment, marked as answer]
Did you mean something like [Laravel Octane](https://laravel.com/docs/9.x/octane)
